ragna/deploy/_ui/components/metadata_filters_builder.py

```diff
@@ -0,0 +1,249 @@
+import panel as pn
+import param
+
+from ragna.core._metadata_filter import MetadataFilter, MetadataOperator
+
+EXCLUDED_OPERATORS = ["RAW", "AND", "OR"]
+ALLOWED_OPERATORS = [
+    op
+    for op in list(MetadataOperator.__members__.keys())
+    if op not in EXCLUDED_OPERATORS
+]
+
+
+class FilterRow(pn.viewable.Viewer):
+    key = param.Selector(objects=[""], default="")
+    operator = param.Selector(objects=[""] + ALLOWED_OPERATORS, default="")
+    value = param.Selector(objects=[""], default="")
+
+    def __init__(self, on_delete_callback, valid_key_value_pairs=None, **params):
+        super().__init__(**params)
+
+        key_select_disabled = True
+        if valid_key_value_pairs:
```

Comment:
User is not allowed to type values, rather they must select from all available options. (Possibly a better UX.)

Reply:
But does `<5` not achieve the same result?